models/productDao.js

```diff
@@ -299,11 +299,132 @@ const getDetailProducts = async (productId) => {
   return detailProduct;
 }
 
+const addProductImages = async(productId, productImageUrl) => {
+  await appDataSource.query(`
+    INSERT INTO product_images (
+      image_url,
+      product_id
+    ) VALUES (?, ?);`,
+    [ productImageUrl, productId ]
+  )
+}
+
+const createProduct = async(userId, name, firstDate, lastDate, price, description, thumbnailImageUrl, participants, dicountRate, scheduleTitle, scheduleEtc, classTypeId, subCategoryId, levelId, locationName, locationLatitude, locationLongitude, locationPlaceUrl, locationGroupName) => {
+
+  const locationGroup = await appDataSource.query(`
+    INSERT INTO location_groups (
+      name
+    ) VALUES (?);`,
+    [ locationGroupName ]
+  )
+
+  const location = await appDataSource.query(`
+    INSERT INTO location (
+      name,
+      latitude,
+      longitude,
+      place_url,
+      location_group_id
+    ) VALUES (?, ?, ?, ?, ?);`,
+    [ locationName, locationLatitude, locationLongitude, locationPlaceUrl, locationGroup.insertId ]
+  )
+
+  const product = await appDataSource.query(`
+  INSERT INTO products (
+    name,
+    first_date,
+    last_date,
+    price,
+    description,
+    thumbnail_image_url,
+    participants,
+    discount_rate,
+    schedule_title,
+    schedule_etc,
+    class_type_id,
+    sub_category_id,
+    user_id,
+    level_id,
+    location_id
+  ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);`,
+  [ name, firstDate, lastDate, price, description, thumbnailImageUrl, participants, dicountRate, scheduleTitle, scheduleEtc, classTypeId, subCategoryId, userId, levelId, location.insertId ]
+  )
+
+  return product.insertId;
+}
+
+const addSchedule = async(productId, schedule) => {
+  
+  const { startTime, finishTime, minutes, content } = schedule
+  
+  await appDataSource.query(`
+    INSERT INTO schedules (
+      start_time,
+      finish_time,
+      minutes,
+      content,
+      product_id
+    ) VALUES (?, ?, ?, ?, ?);`,
+    [ startTime, finishTime, minutes, content, productId ]
+  )
+}
+
+const getProductsList = async (userId) => {
+  const getProductsList = await appDataSource.query(`
+    SELECT
+      p.id AS productId,
+      p.name,
+      p.thumbnail_image_url AS image,
+      p.last_date AS lastDate,
+      s.kor_name AS subCategoryKorName
+    FROM products AS p
+    JOIN sub_categories AS s ON p.sub_category_id = s.id
+    WHERE p.user_id = ?;`,
+    [userId]
+  );
+  return getProductsList;
+};
+
+const deleteProduct = async (userId, productId) => {
+  await appDataSource.query(`
+    DELETE
+    FROM product_images AS p
```

Comment:
Alias (별칭)은 줄이는 본 테이블 명의 의미를 충분히 담고 있어야 한다고 생각합니다.

`pi`정도로 줄여주면 좋을 것 같습니다.
```suggestion
    FROM product_images AS pi
```

Reply:
p -> pi 로 alias 수정하겠습니다!